### PR TITLE
Add PHPCS checks for PHP 8.1 and 8.2

### LINKED ISSUE #5: Add PHP Compatibility Checks for Version 8.1 and 8.2
# User Story

> As a developer, I want to ensure that Elasticsearch Extensions is compatible with all currently supported versions of PHP.

# Acceptance Criteria

- Ensure that Elasticsearch Extensions declares support for PHP 8.1 and 8.2 in composer.json
- Ensure that it states a required minimum of PHP 8.1 in the plugin's metadata
- Ensure that PHPUnit tests are run against PHP 8.1 and 8.2
- Ensure that phpcs compatibility checks are set to check against PHP 8.1, 8.2

### PR DESCRIPTION
**Pull request by moraleida**
Resolves #5 

This updates the plugin configs to enable support for PHP 8.1 and 8.2.
- Supported versions of PHP 8.1 and later
- Supported versions of WordPress 6.2 and later

## Summary by CodeRabbit

- **Documentation**
  - Updated the Continuous Integration (CI) testing documentation to reflect support for PHP versions 8.1 and 8.2, and WordPress version 6.4.

- **Refactor**
  - Enhanced the CI/CD workflows to test against the latest supported PHP and WordPress versions, ensuring compatibility and performance.

- **Chores**
  - Adjusted CI/CD configurations to align with updated coding standards and quality checks for PHP 8.1.

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -9,25 +9,41 @@ jobs:
     strategy:
       fail-fast: false # do not fail fast, let all the failing tests fail.
       matrix:
-        php: [8.0]
+        php: [8.1]
```
[the remark this answers is not in the document]
Comment (moraleida) [reply .github/workflows/unit-tests.yml]: I changed this into another mix of matrix/includes so we get all permutations of php/adapter against the latest ES without multisite, and then 2 significant tests w/ multisite. One for latest php and ES, and two for multisite w/ previous ES.

File: composer.json [hunk 2]
```diff
@@ -56,21 +56,23 @@
...
   ],
+  "require": {
+    "php": "^8.1 || ^8.2"
+  },
   "require-dev": {
-    "php": "^7.4 || ^8.0",
     "alleyinteractive/alley-coding-standards": "^0.3.0",
-    "alleyinteractive/pest-plugin-wordpress": "^0.4.0",
-    "alleyinteractive/searchpress": "^0.4.1",
+    "alleyinteractive/pest-plugin-wordpress": "dev-feature/php-8-1-upgrade",
```
Comment (renatonascalves) [inline composer.json:74]: Suggestion to update to the latest version: https://github.com/alleyinteractive/pest-plugin-wordpress/releases/tag/v0.5.0

**Review by renatonascalves — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (moraleida) [reply composer.json]: @renatonascalves ty! can you give this another look, please?

File: composer.json [hunk 3]
```diff
@@ -56,25 +56,27 @@
...
   ],
+  "require": {
+    "php": "^8.1 || ^8.2"
+  },
   "require-dev": {
-    "php": "^7.4 || ^8.0",
     "alleyinteractive/alley-coding-standards": "^0.3.0",
-    "alleyinteractive/pest-plugin-wordpress": "^0.4.0",
-    "alleyinteractive/searchpress": "^0.4.1",
+    "alleyinteractive/pest-plugin-wordpress": "^0.5.0",
```
Comment (renatonascalves) [inline composer.json:74]: Looks like the xml validation is off.

> Your XML configuration validates against a deprecated schema. Migrate your XML configuration using "--migrate-configuration"!

See [here](https://github.com/alleyinteractive/elasticsearch-extensions/actions/runs/7186725055/job/19572761073?pr=65).

**Review by renatonascalves — APPROVED** [review]
🌵

**Pull request merged**